### ISSUE TITLE
update ktrain to support scikit-learn 0.24.0

### ISSUE THREAD

**Issue by amaiya** [amaiya/ktrain#304]

**Comment by amaiya** [comment]
The release of scikit-learn 0.24.0 requires changing the way some things are imported in **ktrain**.  We have temporarily pinned to `scikit-learn==0.23.2` in ktrain v0.25.3 to prevent errors.  This issue is a reminder to update **ktrain** for `scikit-learn>=v0.24.0` by changing the imports.

**Comment by luismiguells** [comment]
Not yet supported ```scikit-learn>=0.24.0```?

**Comment by sln2737** [comment]
Its almost 2 years now,  I am unable to use Ktrain on 3.9 as the scikit-learn==0.23.2 is not getting through on windows. I think its time to rethink abandoning ktrain.

**Comment by amaiya** [comment]
As of v0.29.1 (just released), **ktrain** is now pinned to `scikit-learn==0.24.2`.  

**Important Note**:    No code changes were necessary for **ktrain** to support scikit-learn 0.24.x.  The only reason it was previously pinned to sklearn 0.23.2 is because of `eli5` used for `TextPredictor.explain`.  The `eli5` tf.keras fork has been updated to support 0.24.2.  So, you just need to update the `eli5` dependency with:
```python
pip uninstall eli5
pip install https://github.com/amaiya/eli5/archive/refs/heads/tfkeras_0_10_1.zip
```
